text-command-line-terminal-applications-and-interfaces.md

```diff
@@ -1,8 +1,25 @@
+Text applications are a class of software ICT that appeared decades ago, prior to the emergence of the graphical user interface (GUI) and the Web. The interface of a text application is generated using only text characters, and either a hardware terminal or a software terminal application handles the rendering of the text application—similar to how a web user agent handles the rendering of a web application. Text applications only accept text input, though some may also support the use of a mouse. More recently, terminal applications which render text applications in the GUI may utilize spoken input through Automated Speech Recognition (ASR) (or other input devices). Both GUI and native text environment interfaces also now commonly support word-completion prediction technologies. Command-line applications are a subset of text applications with further specific properties.
 
-Historically, assistive technologies developed alongside text applications, and several of these use a variety of analysis and scripting techniques to make text applications accessible. Although there are far fewer new text applications being developed compared to new GUI or web applications, text applications remain in use today, and both text applications and the assistive technologies designed for text applications are in active development.
+Historically, assistive technologies developed alongside text applications, and several of these use a variety of analysis and scripting techniques to make text applications accessible. Although there are far fewer new text applications being developed compared to new GUI or web applications, text applications remain in use today. In fact, command-line interfaces have seen a resurgence in recent years, especially in popular programming and revision tracking environments. In some cases this has precipitated renewed developments in  assistive technology support for text applications.
 
-Though this class of applications predates the Web, WCAG can be applied to them. As noted in [Appendix B. Background on Text / Command-line / Terminal Applications and Interfaces](#background-on-text-command-line-terminal-applications-and-interfaces), applying WCAG to text / command-line applications involves understanding how text applications are rendered, how text applications have been made accessible via assistive technologies, and how to apply the concepts of “accessibility supported” and “programmatically determined” to text applications.
+With the exception of closed environments (such as provided by emacs then and now),
+assistive technology support in today's text applications differs markedly
+from that provided users in earlier days. The most marked difference is that
+there's likely less need to make a text-only version of an application
+accessible to AT, because there's almost certainly a graphical version which
+already supports the AX Tree, and likely also a command line version which his
+highly fungible. Key examples  include:
+
```

Comment:
This paragraph has a few issues. The first sentence, talks about Emacs which is a text editor application. It isn't only available in a closed environment, so not sure about the correlation/insertion into the first sentence.  Also not sure about the use of the phrase "then and now" in this sentence.  Emacs has been around since the early days of text-based applications and is still in use today, which is true. Are you saying Emacs also has a GUI version that supports the accessibility tree.  What is meant by "his highly fungible"?  Do you mean "is highly functional"? 

Suggest the following edits:

With the exception of closed environments<del> (such as provided by emacs then and now)</del>, assistive technology support in today's text applications differs markedly from that provided users in earlier days. The most marked difference is that there's likely less need to make a text-only version of an application accessible to AT, because there's almost certainly a <del>graphical</del><ins>GUI</ins> version which already supports the <del>AX Tree</del><ins>accessibility tree or accessibility API</ins>, and likely also a command line version which <del>h</del>is highly <del>fungible</del><ins>comparable to the GUI version (such as that provided by Emacs)</ins>. Key examples  include:

Reply:
In the meeting we accepted my changes.